### PR TITLE
Fixes pointed by static analysis

### PR DESCRIPTION
**Pull request by diegosouza**
This is:

```
- [x] a bugfix
- [ ] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ ] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary

### Why this change is needed?

To have a little less technical debt.

### CODE REVIEW

File: src/PhpSpreadsheet/Reader/IReadFilter.php [hunk 1]
```diff
@@ -9,9 +9,8 @@ interface IReadFilter
      *
      * @param string $column Column address (as a string value like "A", or "IV")
      * @param int $row Row number
-     * @param string $worksheetName Optional worksheet name
      *
      * @return bool
      */
-    public function readCell($column, $row, $worksheetName = '');
+    public function readCell($column, $row);
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Reader/IReadFilter.php]: This is a major breaking change in public interfaces and should be reverted, along with all other implementing classes

File: src/PhpSpreadsheet/Writer/Ods/Content.php [hunk 2]
```diff
@@ -12,7 +12,6 @@
 use PhpOffice\PhpSpreadsheet\Worksheet\Row;
 use PhpOffice\PhpSpreadsheet\Worksheet\Worksheet;
 use PhpOffice\PhpSpreadsheet\Writer\Exception;
-use PhpOffice\PhpSpreadsheet\Writer\Ods;
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Writer/Ods/Content.php]: This is required and must be reverted.

File: src/PhpSpreadsheet/Writer/Xlsx.php [hunk 3]
```diff
@@ -232,6 +232,7 @@ public function save($pFilename)
                         //signed macros ?
                         // Yes : add the certificate file and the related rels file
                         $zip->addFromString('xl/vbaProjectSignature.bin', $this->spreadSheet->getMacrosCertificate());
+                        // @scrutinizer ignore-call
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Writer/Xlsx.php]: A better way to fix this is to introduced dedicated properties for each parts, instead of using a loosely type array. So this should be reverted, and introducing properties for each part should be another dedicated PR.

**Comment by diegosouza** [comment]
> Thank you for the PR, some commits are interesting, but a few of them should be improved before merging. Will you work on that ? or should I cherry-pick only the interesting commits ?

It's better you cherry pick if you want. I won't make additional changes. The codebase is too messy/weak.

**Comment by PowerKiKi** [comment]
Cherry picked what was needed, thanks
